### PR TITLE
Adds more skintones and tweaks existing ones.

### PR DESCRIPTION
**Pull request by SynthTwo**
## What Does This PR Do
First PR, yippee. This PR adds more skintones to choose from, and adjusts 2 of the existing ones to be more accurate. 

Also re-numbered the presets to be from Light to Dark, instead of random. 

## Why It's Good For The Game
More customization good. Having only 8 skintones to choose isn't fun, especially when there's no choice for tanned skin and only 2 of the choices are dark. 

## Images of changes
skins
From left to right: Pale, Coffee, Latte, Bronzed, Caramel, Sienna, Almond. 

image
Comparsion of Coffee and Pale adjustments. Pale was just Default White. Coffee was too desaturated and not realistic. 

Before
numbers

After
image

Shrunk the menu.

## Testing
Spawned in a bunch of flesh people with the new skintones. Stared at them. Looked right. 

## Changelog
:cl:
add: 5 new skintones (Latte, Caramel, Bronzed, Almond, Sienna)
tweak: adjusted 2 skintones (Coffee and Pale)
tweak: re-organized the skintones to be light-dark
/:cl:

### CODE REVIEW

**Comment by Alecksohs** [comment]
Good! By the way, I want people to do this, as the owner of the original TGhuman PR, I want to see more skintones added. I don't think these changes need a wiki update.

**Comment by SynthTwo** [comment]
I shrunk the skin menu a bit, as I realised that when more skins get added eventually, it's just gonna get bigger. 
image

**Comment by SynthTwo** [comment]
Will fix the conflict once I finish university </3

**Comment by SynthTwo** [comment]
I FUCKED UP BAD!

**Comment by SynthTwo** [comment]
the map thing was a big mistake, those labels be inaccurate

File: code/modules/client/preference/link_processing.dm [hunk 1]
```diff
@@ -668,12 +668,7 @@
...
 						var/const/MAX_LINE_ENTRIES = 4
 						var/prompt = "Choose your character's skin tone: 1-[S.icon_skin_tones.len]\n("
-						for(var/i = 1 to S.icon_skin_tones.len)
-							if(i > MAX_LINE_ENTRIES && !((i - 1) % MAX_LINE_ENTRIES))
-								prompt += "\n"
-							prompt += "[i] = [S.icon_skin_tones[i]]"
-							if(i != S.icon_skin_tones.len)
-								prompt += ", "
+						prompt += "Light to Dark"
 						prompt += ")"
```
Comment (lewcc) [inline code/modules/client/preference/link_processing.dm:672]: You can remove the "(" and ")" and just move the "Light to Dark" string inside the prompt string. 

There are a few places that modify skintone like this (mutation_powers.dm, appearance_changer.dm). You can probably go through and modify those to reference this as well if this is what you want to do.

File: code/modules/client/preference/link_processing.dm [hunk 2]
```diff
@@ -668,12 +668,7 @@
 					else if(S.bodyflags & HAS_ICON_SKIN_TONE)
 						var/const/MAX_LINE_ENTRIES = 4
 						var/prompt = "Choose your character's skin tone: 1-[S.icon_skin_tones.len]\n("
```
Comment (lewcc) [inline code/modules/client/preference/link_processing.dm:670]:
```suggestion
						var/prompt = "Choose your character's skin tone: 1-[length(S.icon_skin_tones)]\n("
```

**Review by lewcc — COMMENTED** [review]
This will probably update a bunch of people's skin colors for their character slots, @AffectedArc07 or someone will likely need to run a script to update everyone's colors after this goes in.

**Comment by AffectedArc07** [comment]
> This will probably update a bunch of people's skin colors for their character slots, @AffectedArc07 or someone will likely need to run a script to update everyone's colors after this goes in.

The script for this is gonna need some work as it needs to hold the old values and make a swap for the new ones. 

I think I can do it with a temp table but not 100% sure.

**Comment by SynthTwo** [comment]
Haven't forgotten about this PR, gonna do Lewcc conversation soon 👍😈

**Comment by SynthTwo** [comment]
Will likely follow up adjusting mutation powers (morph) and appearance changer (magic mirror and CMA) in the future. They still function just fine, so them being numbers and names for now is not a big deal, just inconsistent. Discussed this with Lewcc.

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]
PR is good to go but needs AA to merge due to SQL issues

**Comment by AffectedArc07** [comment]
@SynthTwo can you get me a list of what old skin tones map to new ones? Like which old number corresponds to which new number.

**Comment by SynthTwo** [comment]
> @SynthTwo can you get me a list of what old skin tones map to new ones? Like which old number corresponds to which new number.

Sure thing.

**Old Skintones to New Skintones:**

**2 maps to 13 (Chestnut)**
**3 maps to 12 (Coffee)**
**5 maps to 2 (Pale)**
**7 maps to 3 (Classic)**
**8 maps to 5 (Oliver)**

1, 6 and 4 still map to the same numbers. (Default White, Olive, Beige)

The rest are obviously new ones. Sorry if it's a bit confusing.

**Comment by AffectedArc07** [comment]
SQL Update Script

```sql
CREATE TEMPORARY TABLE characters_temp (
id INT,
new_skintone INT
) ENGINE=MEMORY;

INSERT INTO characters_temp (id, new_skintone)
	SELECT
	id,
	CASE
		WHEN skin_tone = 1 THEN 1
		WHEN skin_tone = 2 THEN 13
		WHEN skin_tone = 3 THEN 12
		WHEN skin_tone = 4 THEN 4
		WHEN skin_tone = 5 THEN 2
		WHEN skin_tone = 6 THEN 6
		WHEN skin_tone = 7 THEN 3
		WHEN skin_tone = 8 THEN 5
		ELSE 1
	END AS st
	FROM characters;
	
UPDATE characters, characters_temp
SET characters.skin_tone = characters_temp.new_skintone
WHERE characters.id = characters_temp.id;

DROP TABLE characters_temp;
```

Please dump in make a folder in `tools` called `pr_sql`, then a folder called `22765`, then save the above as `tools/pr_sql/22765/convert.sql`.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**